src/core/pages/structuredMarkdown.tsx

```
@@ -1,19 +1,25 @@
+) {
   const result: React.ReactNode[] = [];
-  const stack: { level: number; content: React.ReactNode[] }[] = [];
+  const stack: { level: number; content: React.ReactNode[]; id: string }[] = [];
   let keyCounter = 0;
+  // Track counters for each parent's children
+  const parentCounters = new Map<string, number>();
 
   const wrapAndPushContent = (level: number) => {
     while (stack.length > 0 && stack[stack.length - 1].level >= level) {
-      const { level: stackLevel, content } = stack.pop()!;
+      const { level: stackLevel, content, id } = stack.pop()!;
```

Biome

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)